### DISCUSSION TITLE
Typescript global data files

### DISCUSSION THREAD

**Discussion by redmagic** [11ty/eleventy#1835] [Q&A]
Hi, I am trying to get a typescript data file to work. My ultimate goal is to expose some helper functions to njk macros via either context or shortcodes.

Things that popped up in my mind:

1. Compile an eleventy.ts file to eleventy.ts
2. Get eleventy to use a typescript loader to `require('somefile.ts')`
3. Create a dataExtension for ts
```js 
eleventyConfig.addDataExtension('ts', async (contents) => {
        // return await { test: () => 456 };
        return await ts.transpileModule(contents).outputText;//this gives us the module, but what next?
    });
```

Has anyone tried this? I am missing some parts of this puzzle, but cannot figure out which.

**Comment by Horsty80** [comment]
Hey,
I'm looking for the same thing, @redmagic have you success something ?

I want to work with typescript file inside my _data 
I'm using it for const and data fetching.

I'm trying doing this by using esbuild to transform ts to js file but it's kind complicated.

Some help would be appreciate :)

**Reply by redmagic** [reply]
Hi Horsty80,

I have not looked into this any further, we have decided to stick with `.js` files.

Kind regards and good luck with yor project.
